### ISSUE TITLE
Job hook not working 

### ISSUE THREAD

**Issue by mksbcisco** [nautobot/nautobot#2337]
### Environment
* Nautobot version : v1.4.1
* Python version: v3.8.10
* Database platform, version: PostgreSQL v12.12
* Middleware(s): n/a

I am trying to execute a job whenever a site is created, to understand what the job is doing and what the parameters are being passed I am returning all the variables. Variables were as used in an example. 
My main objective is to make a REST-API call using POST method to send whatever data has been collected while creation of the site to a remote server where further processing can be done.
Right now, when I create a site the job fails with an error which states that `Job NewSiteReceiver is not enabled to be run!`. I can confirm that the job `NewSiteReceiver` is enabled in `Job hook` section.

Snippet of the code is below for reference:
`
from django.utils.text import slugify
from nautobot.dcim.models import Device, DeviceRole, DeviceType, Manufacturer, Site
from nautobot.extras.models import Status
from nautobot.extras.jobs import *
from nautobot.extras.choices import ObjectChangeActionChoices
from nautobot.extras.jobs import JobHookReceiver

class NewSiteReceiver(JobHookReceiver):

    def receive_job_hook(self, change, action, changed_object):
        # return on delete action
        if action == ObjectChangeActionChoices.ACTION_CREATE:
            return self,change,action,changed_object
`

**Comment by glennmatthews** [comment]
Thanks for the report. Can you confirm that the job is also enabled under `Jobs`? (Filter on "Is job hook receiver: Yes" so that the jobhookreceiver job appears in the list). Trying to figure out if we have a functional bug here or a documentation gap.

**Comment by mksbcisco** [comment]
@glennmatthews So my impression was that "Job Hooks" are standalone and can action as a job themselves. 
Having said that there is nothing under "Jobs" 
The "Job Hook" is enabled

**Comment by mksbcisco** [comment]
Yes the job ran I am just trying to figure out now how to make the actual API call and send the whole change data to the endpoint. In case you have time please do guide on it or give a snippet of a code that I could follow. Thanks

**Comment by lampwins** [comment]
@mksbcisco we do have this example in the docs that shows how you can access the snapshot data from the `ObjectChange` instance. https://nautobot.readthedocs.io/en/v1.4.1/models/extras/jobhook/#example-job-hook-receiver

We will take an action on our side that we should better document the `ObjectChange` interface.
